cnd/src/btsieve/client.rs

```diff
@@ -214,13 +222,51 @@ impl BtsieveHttpClient {
         Box::new(
             self.client
                 .delete(query.as_ref().clone())
+                .headers(construct_headers())
                 .send()
                 .map(|_| ())
                 .map_err(|e| {
                     Error::FailedRequest(format!("Failed to delete query because {:?}", e))
                 }),
         )
     }
+
+    pub fn health(&self) -> Box<dyn Future<Item = (), Error = ()> + Send> {
+        let mut url = self.endpoint.clone();
+        url.set_path("health");
+
+        let btsieve_endpoint = self.endpoint.clone();
+        Box::new(
+            self.client
+                .get(url.clone())
+                .headers(construct_headers())
+                .send()
+                .map(|response| {
+                    let endpoint = &response.url()[Position::BeforeScheme..Position::BeforePath];
+
+                    if response.status().is_success() {
+                        log::info!("Btsieve running at {}", endpoint)
+                    } else {
+                        log::error!(
+                            "Version of btsieve at {} does not match expected version",
```

Comment:
From this code it is not immediately apparent why this error message is what it is i.e., the code does not show _why_ this error is caused by a version mismatch.

Reply:
I agree. I'm not too happy with the logs. I should probably just log the HTTP API Problem returned by btsieve. The bot merged this before I could do it though, so it'll have to be in a follow-up PR.